OS version in the library description is ignored

Tried with Amidst in jar version with version : 3.6, 3.7, 4.0, 4.1, 4.2.beta1 and 4.2.beta2 and I have always the same result : it works with MC1.9.4 or MC1.8.9 but crashes with MC1.5.2 or MC1.4.7 :

```
Amidst has encountered an uncaught exception on the thread Thread[Thread-6,1,main]
[info] Amidst v4.1
[info] Current system time: 2016-05-17 15:33:05.537
[info] System.getProperty("os.name") == 'Mac OS X'
[info] System.getProperty("os.version") == '10.8.5'
[info] System.getProperty("os.arch") == 'x86_64'
[info] System.getProperty("java.version") == '1.8.0_73'
[info] System.getProperty("java.vendor") == 'Oracle Corporation'
[info] System.getProperty("sun.arch.data.model") == '64'
[info] Enabling OpenGL.
[info] Biome profile activated: default
[info] using '.minecraft' directory at: '/Users/2frei/Library/Application Support/minecraft', libraries: '/Users/2frei/Library/Application Support/minecraft/libraries'
[info] using biome profiles at: 'biome'
[info] Scanning for profiles.
[info] Successfully loaded profile list.
[info] Beginning latest version list load.
[info] Attempting to download remote version list…
[warning] cannot find valid profile directory for launcher profile 'MapWriter': /Volumes/Mac_Docs/2frei/Jeux/Minecraft/Profiles/MapWriter
[info] Successfully loaded version list. URL: https://launchermeta.mojang.com/mc/game/version_manifest.json
[warning] cannot find valid version directory for launcher profile 'Cubenis'
[warning] cannot find valid version directory for launcher profile 'Epicube'
[info] using minecraft launcher profile 'test' with versionId '1.5.2'
[info] Loading libraries.
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/minecraft/launchwrapper/1.5/launchwrapper-1.5.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/sf/jopt-simple/jopt-simple/4.5/jopt-simple-4.5.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/org/ow2/asm/asm-all/4.1/asm-all-4.1.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar
[info] Skipping library: org.lwjgl.lwjgl:lwjgl:2.9.0
[info] Skipping library: org.lwjgl.lwjgl:lwjgl_util:2.9.0
[info] Skipping library: org.lwjgl.lwjgl:lwjgl-platform:2.9.0
[warning] Failed attempt to load library at: /Users/2frei/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl/2.9.1-nightly-20130708-debug3
[info] Skipping library: org.lwjgl.lwjgl:lwjgl:2.9.1-nightly-20130708-debug3
[warning] Failed attempt to load library at: /Users/2frei/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl_util/2.9.1-nightly-20130708-debug3
[info] Skipping library: org.lwjgl.lwjgl:lwjgl_util:2.9.1-nightly-20130708-debug3
[warning] Failed attempt to load library at: /Users/2frei/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl-platform/2.9.1-nightly-20130708-debug3
[info] Skipping library: org.lwjgl.lwjgl:lwjgl-platform:2.9.1-nightly-20130708-debug3
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-osx.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-osx.jar
[crash] Amidst has encounted an uncaught exception on the thread Thread[Thread-6,1,main]
[crash] java.lang.NoClassDefFoundError: org/lwjgl/LWJGLException
    at java.lang.Class.getDeclaredFields0(Native Method)
    at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
    at java.lang.Class.getDeclaredFields(Class.java:1916)
    at amidst.mojangapi.minecraftinterface.RecognisedVersion.getMainClassFields(RecognisedVersion.java:133)
    at amidst.mojangapi.minecraftinterface.RecognisedVersion.generateMagicString(RecognisedVersion.java:126)
    at amidst.mojangapi.minecraftinterface.RecognisedVersion.from(RecognisedVersion.java:120)
    at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterfaceBuilder.create(LocalMinecraftInterfaceBuilder.java:34)
    at amidst.mojangapi.file.directory.VersionDirectory.createLocalMinecraftInterface(VersionDirectory.java:106)
    at amidst.mojangapi.MojangApi.set(MojangApi.java:74)
    at amidst.gui.profileselect.LocalProfileComponent.tryLoad(LocalProfileComponent.java:82)
    at amidst.threading.WorkerExecutor.lambda$run$11(WorkerExecutor.java:69)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.lwjgl.LWJGLException
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 14 more
```

I deleted everything in the .minecraft folder
I launched MC to DL a proper version of the launcher
I created one profile and just modified the use of version 1.5.2
I launched MC, generated a new world and just cut a tree
I saved and quitted MC
Start amidst and it crashed (see below).

Maybe it's due to OSX, I will try with java 1.8.0_91

`[info] Amidst v4.2-beta2
[info] Current system time: 2016-05-19 18:16:16.425
[info] System.getProperty("os.name") == 'Mac OS X'
[info] System.getProperty("os.version") == '10.8.5'
[info] System.getProperty("os.arch") == 'x86_64'
[info] System.getProperty("java.version") == '1.8.0_73'
[info] System.getProperty("java.vendor") == 'Oracle Corporation'
[info] System.getProperty("sun.arch.data.model") == '64'
[info] Enabling OpenGL.
[info] Biome profile activated: default
[info] using '.minecraft' directory at: '/Users/2frei/Library/Application Support/minecraft', libraries: '/Users/2frei/Library/Application Support/minecraft/libraries'
[info] using biome profiles at: 'biome'
[info] Scanning for profiles.
[info] Successfully loaded profile list.
[info] Beginning latest version list load.
[info] Attempting to download remote version list...
[info] Successfully loaded version list. URL: https://launchermeta.mojang.com/mc/game/version_manifest.json
[info] using minecraft launcher profile 'test5' with versionId '1.5.2'
[info] Loading libraries.
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/minecraft/launchwrapper/1.5/launchwrapper-1.5.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/sf/jopt-simple/jopt-simple/4.5/jopt-simple-4.5.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/org/ow2/asm/asm-all/4.1/asm-all-4.1.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jinput/jinput/2.0.5/jinput-2.0.5.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jutils/jutils/1.0.0/jutils-1.0.0.jar
[info] Skipping library: org.lwjgl.lwjgl:lwjgl:2.9.0
[info] Skipping library: org.lwjgl.lwjgl:lwjgl_util:2.9.0
[info] Skipping library: org.lwjgl.lwjgl:lwjgl-platform:2.9.0
[warning] Failed attempt to load library at: /Users/2frei/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl/2.9.1-nightly-20130708-debug3
[info] Skipping library: org.lwjgl.lwjgl:lwjgl:2.9.1-nightly-20130708-debug3
[warning] Failed attempt to load library at: /Users/2frei/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl_util/2.9.1-nightly-20130708-debug3
[info] Skipping library: org.lwjgl.lwjgl:lwjgl_util:2.9.1-nightly-20130708-debug3
[warning] Failed attempt to load library at: /Users/2frei/Library/Application Support/minecraft/libraries/org/lwjgl/lwjgl/lwjgl-platform/2.9.1-nightly-20130708-debug3
[info] Skipping library: org.lwjgl.lwjgl:lwjgl-platform:2.9.1-nightly-20130708-debug3
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-osx.jar
[info] Found library: /Users/2frei/Library/Application Support/minecraft/libraries/net/java/jinput/jinput-platform/2.0.5/jinput-platform-2.0.5-natives-osx.jar
[crash] Amidst has encounted an uncaught exception on the thread Thread[Thread-6,1,main]
[crash] java.lang.NoClassDefFoundError: org/lwjgl/LWJGLException
	at java.lang.Class.getDeclaredFields0(Native Method)
	at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
	at java.lang.Class.getDeclaredFields(Class.java:1916)
	at amidst.mojangapi.minecraftinterface.RecognisedVersion.getMainClassFields(RecognisedVersion.java:130)
	at amidst.mojangapi.minecraftinterface.RecognisedVersion.generateMagicString(RecognisedVersion.java:124)
	at amidst.mojangapi.minecraftinterface.RecognisedVersion.from(RecognisedVersion.java:119)
	at amidst.mojangapi.minecraftinterface.local.LocalMinecraftInterfaceBuilder.create(LocalMinecraftInterfaceBuilder.java:32)
	at amidst.mojangapi.file.directory.VersionDirectory.createLocalMinecraftInterface(VersionDirectory.java:99)
	at amidst.mojangapi.MojangApi.set(MojangApi.java:71)
	at amidst.gui.profileselect.LocalProfileComponent.tryLoad(LocalProfileComponent.java:84)
	at amidst.threading.WorkerExecutor.lambda$run$11(WorkerExecutor.java:71)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.lwjgl.LWJGLException
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 14 more
`

--- a/src/main/java/amidst/mojangapi/file/LibraryFinder.java
+++ b/src/main/java/amidst/mojangapi/file/LibraryFinder.java
@@ -38,7 +38,7 @@ public enum LibraryFinder {
 
 	private static File getLibraryFile(File librariesDirectory, LibraryJson library) {
 		try {
-			if (library.isActive(getOs())) {
+			if (library.isActive(getOs(), OperatingSystemDetector.getVersion())) {
 				return getLibraryFile(getLibrarySearchPath(librariesDirectory, library.getName()));
 			} else {
 				return null;

--- a/src/main/java/amidst/mojangapi/file/json/version/LibraryJson.java
+++ b/src/main/java/amidst/mojangapi/file/json/version/LibraryJson.java
@@ -24,13 +24,13 @@ public class LibraryJson {
 	 * applicable rule. However, this might be wrong so we need to take the most
 	 * specific rule? For now this works fine.
 	 */
-	public boolean isActive(String os) {
+	public boolean isActive(String os, String version) {
 		if (rules.isEmpty()) {
 			return true;
 		}
 		boolean result = false;
 		for (LibraryRuleJson rule : rules) {
-			if (rule.isApplicable(os)) {
+			if (rule.isApplicable(os, version)) {
 				result = rule.isAllowed();
 			}
 		}

--- a/src/main/java/amidst/mojangapi/file/json/version/LibraryRuleJson.java
+++ b/src/main/java/amidst/mojangapi/file/json/version/LibraryRuleJson.java
@@ -14,8 +14,8 @@ public class LibraryRuleJson {
 	public LibraryRuleJson() {
 	}
 
-	public boolean isApplicable(String os) {
-		return this.os == null || this.os.getName().equals(os);
+	public boolean isApplicable(String os, String version) {
+		return this.os == null || this.os.isApplicable(os, version);
 	}
 
 	public boolean isAllowed() {

--- a/src/main/java/amidst/mojangapi/file/json/version/LibraryRuleOsJson.java
+++ b/src/main/java/amidst/mojangapi/file/json/version/LibraryRuleOsJson.java
@@ -1,17 +1,20 @@
 package amidst.mojangapi.file.json.version;
 
+import java.util.regex.Pattern;
+
 import amidst.documentation.GsonConstructor;
 import amidst.documentation.Immutable;
 
 @Immutable
 public class LibraryRuleOsJson {
 	private volatile String name;
+	private volatile String version;
 
 	@GsonConstructor
 	public LibraryRuleOsJson() {
 	}
 
-	public String getName() {
-		return name;
+	public boolean isApplicable(String os, String version) {
+		return this.name.equals(os) && (this.version == null || Pattern.matches(this.version, version));
 	}
 }

--- a/src/main/java/amidst/util/OperatingSystemDetector.java
+++ b/src/main/java/amidst/util/OperatingSystemDetector.java
@@ -7,6 +7,7 @@ public enum OperatingSystemDetector {
 	;
 
 	private static String OS_NAME = System.getProperty("os.name").toLowerCase();
+	private static String OS_VERSION = System.getProperty("os.version");
 
 	public static boolean isWindows() {
 		return OS_NAME.indexOf("win") >= 0;
@@ -22,5 +23,9 @@ public enum OperatingSystemDetector {
 
 	public static boolean isSolaris() {
 		return OS_NAME.indexOf("sunos") >= 0;
+	}
+
+	public static String getVersion() {
+		return OS_VERSION;
 	}
 }

